api/client/service.proto

```diff
@@ -34,10 +36,62 @@ message ExecutionResponse {
   google.rpc.Status error_detail = 7;
 }
 
+// Used to request a RequestStreamResponse.
+message RequestStreamRequest {
+  // Indicates the amount of RequestStreamResponses that should be yielded by the server
+  // in response to this request.
+  // Used to implement basic flow control by the client. Once AsyncClientImpl gets a way
+  // to apply backpressure this could probably be dropped.
```

Comment:
Non-actionable, but one thing you'll need to watch out for is potential coordination between Nighthawk when it's under load and backed up and the gRPC request stream. There may be situations in which we're applying backpressure that somehow related to the system-under-test. Not sure how likely this is, probably the best practice is to always ensure that Nighthawk itself is not max loaded, but is kept to something like 50% or 80% of CPU.

Reply:
Good point; putting some do's and dont's out there in documentation will surely be helpful. Cross linking #241 so we'll remember this particular comment. It might also be a good practice to use different network cards for sourcing data and replaying.  Also leaving a tangentially related thought here so I don't forget: hypothetically the replay service might not keep up. In the WIP PR associated to this that will be tracked as blocking. This might be either valid (the gRPC data source implements a poor-mans version of request-release timing, or problematic when it's not intentional). This should be addressed at the `RequestSource` level as it's not specific to gRPC. Possibly a special `--rps` value to enable users to explicitly express intent for `as fast as possible` would help here, as we can then discern between problematic and (probably) intentional scenarios.